src/compression/GzipCompression.cpp

```diff
@@ -0,0 +1,107 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2017 Intel Corporation. All Rights Reserved.
+
+#include <iostream>
+#include <cstring>
+#include <zlib.h>
+#include "GzipCompression.h"
+#include <ipDeviceCommon/Statistic.h>
+
+GzipCompression::GzipCompression(int t_width, int t_height, rs2_format t_format, int t_bpp)
+{
+        m_format = t_format;
+        m_width = t_width;
+        m_height = t_height;
+        m_bpp = t_bpp;
+        m_windowsBits = 15;
+        m_gzipEncoding = 16;
+}
+
+int GzipCompression::compressBuffer(unsigned char *t_buffer, int t_size, unsigned char *t_compressedBuf)
+{
+#ifdef STATISTICS
+        Statistic::getStatisticStreams()[rs2_stream::RS2_STREAM_DEPTH]->m_compressionBegin = std::chrono::system_clock::now();
+#endif
+        int compressedSize = 0;
+        m_strm.zalloc = Z_NULL;
+        m_strm.zfree = Z_NULL;
+        m_strm.opaque = Z_NULL;
+        m_strm.next_in = (Bytef *)t_buffer;
+        m_strm.avail_in = t_size;
+        m_strm.next_out = (Bytef *)t_compressedBuf + sizeof(compressedSize);
+        m_strm.avail_out = t_size;
+        int z_result = deflateInit2(&m_strm, Z_BEST_SPEED /*Z_DEFAULT_COMPRESSION*/, Z_DEFLATED, m_windowsBits | m_gzipEncoding, 8, Z_DEFAULT_STRATEGY);
+        if (z_result != Z_OK)
+        {
+                printf("error: init frame compression with gzip failed\n");
```

Comment:
printf --> log

Reply:
Currently we do not have log infrastructure, it's a high priority task, we'll do it ASAP

Reply:
Done